fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/F-PLN/MfdFmsFplnArr.tsx

```diff
@@ -454,9 +455,9 @@ export class MfdFmsFplnArr extends FmsPage<MfdFmsFplnArrProps> {
                     sec: this.secActive,
                   }}
                 >
-                  {this.rwyLength}
+                  {NXUnits.mToUser(this.rwyLength)}
```

Comment:
**question:** I'm not sure if I'm missing something, but I don't think this works. `NXUnits.mToUse` takes in a `number`, and you are passing in a `Subject<number>`.

I think we need to approach this differently. Even if this worked, right now, it would not react to the EFB setting changing, potentially requiring restarting your flight.

I would suggest looking into the `NumberUnit` / `NumberUnitSubject` types from `msfs-sdk`, and properly hooking that to react with the kg/lbs setting from `NXDataStore`.

Reply:
- That is correct, have made a correction to it.

in regards to the approach, it does work to do the conversion directly in the onNewData() method inside the component.

It does not however work in real-time with the EFB setting changing, i.e you need to reload the entire page, by going out of it and into it again, **however this is the case with all the other NXUnit conversions as well.** so if it has to happen in real time, then it needs to be used for all the areas.

Reply:
I don't think that would be very nice. A better approach might be something wrapping or inheriting NumberUnitSubject and that can hide all the details of listening for the setting, and performing the conversions.

Reply:
If anything, it would be subscribing to NXDataStore out in MFDPageDirectory and the passing that value down, so everything refreshes when it changes. Then we never need to ask if its metric on NXUnit, and we can then make etc. `NumberUnit(value, KG).asUnit(passedDownUnit)` and call it a day.

Reply:
The UI does not react to things that are not `Subscribable` values. I'm not sure what you mean. So there is value in `NumberUnitSubject`.

It also even has its own `asUnit` that takes a subscribable of a unit - allowing you to refresh the UI when the unit changes.